src/java/org/apache/cassandra/service/StartupChecks.java

```diff
@@ -285,6 +286,49 @@ public void execute()
         }
     };
 
+    public static final StartupCheck checkReadAheadKbSetting = new StartupCheck()
+    {
+        // This value is in KB.
+        private static final long EXPECTED_READ_AHEAD_KB_SETTING = 8;
+        private static final String READ_AHEAD_KB_SETTING_PATH = "/sys/block/sda/queue/read_ahead_kb";
```

Comment:
Probably should be checked not only on `sda` device, but on all data directories (not sure that it's reasonable to check `read_ahead_kb` for commit log directory)
It's common to set more than one directory in `data_file_directories` property in `cassandra.yaml`
```
data_file_directories:
  - /srv/disk01
  - /srv/disk02
```

`lsblk` could be used to find mount point on common hardware (and to match device name with directory)
```
$ lsblk
NAME                         MAJ:MIN RM   SIZE RO TYPE  MOUNTPOINT
sda                            8:0    0   1.5T  0 disk  /srv/disk01
sdb                            8:16   0   1.5T  0 disk  
`-md0                          9:0    0   5.8T  0 raid0 
  `-md0p1                    259:2    0   5.8T  0 part  /srv/disk02
...
```
but pay special attention for case with software raid, where mount point could be on raid partition like on example above
software raid also could have it's own `read_ahead_kb` and makes sense to check it too.

Reply:
yeah ideally we'll want to go through all data dirs and dynamically find the underlying devices to be able to lookup the `read_ahead_kb` setting. I'd like to avoid using `lsblk` and use something native to java for this check if possible (ie. [Files.getFileStore](https://docs.oracle.com/javase/7/docs/api/java/nio/file/Files.html#getFileStore(java.nio.file.Path)). Ideally we would want to do a manual test with this with multiple data dirs and a raid system like in your example to make sure the test is working. Do you have acess to a box where you can test this?

Reply:
Just to clarify the steps.
1) Iterate through the data directories.
2) Get the volume using Files.getFileStore() for the directory, Example it returns /dev/sda2 for /home/<user>/cassandra/data
3) Read the path ignoring the numbers from the volume returned in step 2 -  /sys/block/**sda**/queue/read_ahead_kb

Reply:
yep, looks good, with some clarification: after step 2 collect all paths (without numbers) into `Set` before going to step 3, in order to avoid duplicate messages when two data directories points onto the same underlying device

Reply:
sounds good 👍

Reply:
Added logic, thanks guys.